Firestore/core/src/firebase/firestore/remote/datastore.mm

```diff
@@ -148,23 +148,23 @@ void LogGrpcCallFinished(absl::string_view rpc_name,
 
 void Datastore::CommitMutations(NSArray<FSTMutation*>* mutations,
                                 FSTVoidErrorBlock completion) {
-  grpc::ByteBuffer message = serializer_bridge_.ToByteBuffer(
-      serializer_bridge_.CreateCommitRequest(mutations));
-
   ResumeRpcWithCredentials(
-      [this, message, completion](const StatusOr<Token>& maybe_credentials) {
+      [this, mutations, completion](const StatusOr<Token>& maybe_credentials) {
         if (!maybe_credentials.ok()) {
           completion(util::MakeNSError(maybe_credentials.status()));
           return;
         }
-        CommitMutationsWithCredentials(maybe_credentials.ValueOrDie(), message,
-                                       completion);
+        CommitMutationsWithCredentials(maybe_credentials.ValueOrDie(),
+                                       mutations, completion);
       });
 }
 
 void Datastore::CommitMutationsWithCredentials(const Token& token,
-                                               const grpc::ByteBuffer& message,
+                                               NSArray<FSTMutation*>* mutations,
                                                FSTVoidErrorBlock completion) {
+  grpc::ByteBuffer message = serializer_bridge_.ToByteBuffer(
```

Comment:
If grpc has already shut down at this point, what does this do? (Or is that even possible to get here if grpc has already shut down?)

Reply:
gRPC isn't shut down until `Datastore` is shut down, and [this check](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/remote/datastore.mm#L272) should prevent this function from being invoked if `Datastore` is shut down.

Reply:
Ah, excellent. lgtm.